rubrik_cdm/cluster.py

```diff
@@ -529,3 +529,124 @@ def refresh_vcenter(self, vcenter_ip, wait_for_completion=True, timeout=15):
+        """Update the proxy configuration on the Rubrik cluster.
+
+        Arguments:
+            host {str} -- The IP address or FQDN of the proxy you wish to add.
+            protocol {str} -- The protocol of the proxy you wish to add.
+            port {int} -- The port of the proxy you wish to add.
+
+        Keyword Arguments:
+            username {str} -- The proxy username used for authentication. (default: {None})
+            password {str} -- The proxy password used for authentication. (default: {None})
+            timeout {int} -- The number of seconds to wait to establish a connection the Rubrik cluster before returning a timeout error. (default: {15})
+
+        Returns:
+            dict -- The full API response for `PATCH /node_management/proxy_config`
+        """
+
+        valid_protocols = [
+            'HTTP',
+            'HTTPS',
+            'SOCKS5']
+
+        if protocol not in valid_protocols:
+            sys.exit("Error: The protocol argument must be one of the following: {}.".format(
+                valid_protocols))
+
+        config = {}
+        config["host"] = host
+        config["protocol"] = protocol
+        config["port"] = port
+        config["username"] = username
+        config["password"] = password
+
+        self.log("proxy: Updated proxy configuration.")
+        return self.patch("internal", "/node_management/proxy_config", config, timeout)
+
+    def delete_proxy(self, timeout=15):
+        """Delete the proxy configuration from the Rubrik cluster.
+
+        Keyword Arguments:
+            timeout {int} -- The number of seconds to wait to establish a connection the Rubrik cluster before returning a timeout error. (default: {15})
+
+        Returns:
+            dict -- The full API response for `DELETE /node_management/proxy_config`
+        """
+
+        current_proxy_settings = self.get("internal", "/node_management/proxy_config", timeout=timeout)
+
+        if current_proxy_settings["host"] == "":
+            return "No change required. The proxy configuration is already cleared out."
+
+        self.log("proxy: Deleted proxy configuration.")
+        return self.delete("internal", "/node_management/proxy_config", timeout)
+
+    def add_guest_credential(self, username, password, domain=None, timeout=15):
+        """Add a new guest credential to the Rubrik cluster.
+
+        Arguments:
+            username {str} -- The account username used for authentication.
+            password {str} -- The account password used for authentication.
+
+        Keyword Arguments:
+            domain {int} -- The domain name of account password used for authentication. (default: {None})
+            timeout {int} -- The number of seconds to wait to establish a connection the Rubrik cluster before returning a timeout error. (default: {15})
+
+        Returns:
+            dict -- The full API response for `POST /vmware/guest_credential`
```

Comment:
add the API version to the return value

Reply:
Is this what you are suggesting?

`dict -- The full API response for `POST /v1/vmware/guest_credential``